broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -188,12 +190,22 @@ private void leaderTransition(final long newTerm) {
         .onComplete(
             (success, error) -> {
               if (error == null) {
-                partitionListeners.forEach(
-                    l -> l.onBecomingLeader(partitionId, newTerm, logStream));
+                partitionListeners.stream()
+                    .map(l -> l.onBecomingLeader(partitionId, newTerm, logStream))
+                    .forEach(
+                        future ->
+                            future.onComplete(
```

Comment:
Btw I think the actor scheduler support a method which accepts multiple future to react on a failure